### PR TITLE
App distribution signin persist

### PR DESCRIPTION
**Pull request by schnecle**
Store a boolean if the tester successfully signs in and registers a FID.
Tested by running Bee Plus locally.

### CODE REVIEW

File: FirebaseAppDistribution/Sources/FIRAppDistribution.m [hunk 1]
```diff
@@ -22,8 +22,10 @@
 #import "FIRAppDistributionMachO+Private.h"
 #import "FIRAppDistributionRelease+Private.h"
 #import "FIRFADApiService+Private.h"
+#import "FIRFADLocalStorage+Private.h"
 #import "FIRFADLogger+Private.h"
 
+
```
[the remark this answers is not in the document]
Comment (schnecle) [reply FirebaseAppDistribution/Sources/FIRAppDistribution.m:28]: yeah, need to run all the style stuff again :-)

File: FirebaseAppDistribution/Sources/FIRFADLocalStorage.m [hunk 2]
```diff
@@ -13,24 +13,22 @@
...
-- (instancetype)init {
-  self = [super init];
-
-  return self;
-}
-
-- (BOOL)isTesterSignedIn {
-  return YES;
++ (BOOL)isTesterSignedIn {
+  NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
```
Comment (maksymmalyhin) [inline FirebaseAppDistribution/Sources/FIRFADLocalStorage.m:20]: FYI, we experienced thread-safety issues with `NSUserDefaults` on some iOS version, so we had to opt out for a custom thread-safe implementation [GULUserDefaults](https://cs.opensource.google/firebase-sdk/firebase-ios-sdk/+/master:GoogleUtilities/UserDefaults/Private/GULUserDefaults.h). It modifies the same user default but via C API under the hood. You may consider using  `GULUserDefaults` if you access it outside of main thread.

Comment (jeremydurham) [reply FirebaseAppDistribution/Sources/FIRFADLocalStorage.m:20]: Good point @maksymmalyhin! We talked about this offline and will likely switch to using GULUserDefaults

Comment (schnecle) [reply FirebaseAppDistribution/Sources/FIRFADLocalStorage.m]: Done!

File: FirebaseAppDistribution/Sources/FIRAppDistribution.m [hunk 3]
```diff
@@ -174,18 +190,7 @@ - (NSString *)getAppName {
...
-  //  if (!didClearAuthState) {
-  //    FIRFADErrorLog(@"Error clearing token from keychain: %@", [error localizedDescription]);
-  //    [self logUnderlyingKeychainError:error];
-  //
-  //  } else {
-  //    FIRFADInfoLog(@"Successfully cleared auth state from keychain");
-  //  }
-
-  self.isTesterSignedIn = false;
+  return [FIRFADLocalStorage registerSignOut];
```
[the remark this answers is not in the document]
Comment (schnecle) [reply FirebaseAppDistribution/Sources/FIRAppDistribution.m]: done.

File: FirebaseAppDistribution/Sources/FIRFADLocalStorage.m [hunk 4]
```diff
@@ -13,24 +13,22 @@
...
++ (void)registerSignIn {
+  NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
+  [userDefaults setBool:YES forKey:kSignInStateKey];
 }
 
-- (BOOL)clearSignInState:(NSError *_Nullable)error {
-  return YES;
++ (void)registerSignOut {
+  NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
+  [userDefaults setBool:YES forKey:kSignInStateKey];
```
[the remark this answers is not in the document]
Comment (schnecle) [reply FirebaseAppDistribution/Sources/FIRFADLocalStorage.m]: the intent was to make a distinction at least internally between "I have never seen this before" and "I have registered and explicitly said I want to sign out" - we can decide if we want to change this later.

**Pull request merged**